Automattic/woocommerce-payments
# Fix Afterpay checkout error when shipping information is missing

Fixes #7509 

#### Changes proposed in this Pull Request

- Validate shipping address from order and fallback to billing address if it is missing or invalid.









#### Testing instructions





* Merchant: Disable shipping in you local store: go to  WooCommerce > Settings > General and set 'Shipping location(s)' as 'Disable shipping& shipping calculations'
* Shopper: add product to cart and proceed to checkout
* Shopper: fill the checkout form and place the order paying with Afterpay
* Observe there are no errors and the checkout proceeds to the Stripe test page.
* Check the shipping data matches the **billing** information provided in previous step.
* Authorize Payment and check that you are redirected back to the order received page.
* Merchant: Enable shipping in your local store (i.e. Ship to all countries)
* Shopper: add product to cart and proceed to checkout
* Shopper: fill the checkout form, select 'Ship to different address' and fill the shipping address details. 
* Shopper: place order.
* Observe there are no errors and the checkout proceeds to the Stripe test page.
* Check the shipping data matches the **shipping** information provided in previous step.
* Authorize Payment and check that you are redirected back to the order received page.

-------------------


**Post merge**
